src/classes/PMT_Payment_TDTM.cls

```diff
@@ -44,18 +103,51 @@ public class PMT_Payment_TDTM extends TDTM_Runnable {
     * @param objResult the describe for Opportunity 
     * @return dmlWrapper. 
     ********************************************************************************************************/
-    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
-    TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
-
-        map<Id, Opportunity> mapIdOppNew;
-        map<Id, Opportunity> mapIdOppOld;
-        if (newlist != null) 
-           mapIdOppNew = new map<id, Opportunity>((list<Opportunity>) newlist);
-        if (oldlist != null) 
-           mapIdOppOld = new map<id, Opportunity>((list<Opportunity>) oldlist);
-                       
-        DmlWrapper dmlWrapper = new dmlWrapper();
-        PMT_PaymentCreator pc = new PMT_PaymentCreator(mapIdOppNew, mapIdOppOld, triggerAction, dmlWrapper); 
+    public static DmlWrapper runForOpportunities(List<Opportunity> newlist, List<Opportunity> oldlist,
+            TDTM_Runnable.Action triggerAction) {
+
+        Map<Id, Opportunity> mapIdOppNew = (newlist != null ? new Map<Id, Opportunity>(newlist) : new Map<Id, Opportunity>());
+        Map<Id, Opportunity> mapIdOppOld = (oldlist != null ? new Map<Id, Opportunity>(oldlist) : new Map<Id, Opportunity>());
+
+        DmlWrapper dmlWrapper = new DmlWrapper();
+
+        // Create new Payments as necessary
+        PMT_PaymentCreator pc = new PMT_PaymentCreator(mapIdOppNew, mapIdOppOld, triggerAction, dmlWrapper);
+
+        //if currency has changed, reset payment currencies
```

Comment:
is this logic that really should be in PMT_PaymentCreator() instead?

Reply:
It's legacy logic (and method), but in general I think so. That class handles creating and updating payment records as necessary from the Opportunity.